nessie/src/main/java/org/apache/iceberg/nessie/NessieCatalog.java

```diff
@@ -194,7 +195,7 @@ protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
         ContentKey.of(
             org.projectnessie.model.Namespace.of(tableIdentifier.namespace().levels()),
             tr.getName()),
-        client.withReference(tr.getReference(), tr.getHash()),
+        refreshedClient().withReference(tr.getReference(), tr.getHash()),
```

Comment:
I believe in this case, since the client goes into `NessieTableOperations`, it's reference should be loaded immediately and made immutable regardless of whether the commit hash is specified or not. Currently, the reference will be loaded on first use, which may happen a lot later. WDYT?

Reply:
Actually, I think I was confused. Current code LGTM in this case.